request.py

```diff
@@ -0,0 +1,127 @@
+@auth.login_required
+def get_resource():
+    return jsonify({'data': 'Hello, %s!' % g.user.username})
+
+
+@app.route('/User/<username>/Game', methods=['POST', 'GET'])
+def create_game(username):
+    logging.info(f"User logged in as: {username}")
+    if request.method == 'POST':
+        try:
+            function.create_new_game(username=username)  # How is this parameter supposed to be set?
+            return make_response(jsonify(response='OK'), 201)
+        except sqlalchemy.exc.InvalidRequestError:
+            return make_response(jsonify(error='Bad request'), 400)
+    if request.method == 'GET':
+        new_game = function.get_new_game(username=username)
+        return jsonify(new_game.serialize())
+
+
+@app.route('/User/<username>/Game/<secret_word>/<is_done>', methods=['GET', 'PATCH'])
+def update_retrieve_game(username, secret_word, is_done):
+    logging.info(f'User has given input secret_word as: {secret_word}')
+    logging.info(f'User has given input username as: {username}')
+
+    game_result = function.get_game(username, secret_word)
+    if request.method == 'GET':
+        return jsonify(game_result.serialize())  # I don't know if I need to present them serialized to my gameplay..
+
+    if request.method == 'PATCH':
+        data = request.json
+        is_done = data["is_done"]
+        if is_done:
+            try:
+                fields = {'got_hint': data["got_hint"], 'missed_letters': data["missed_letters"], 'correct_letters': data["correct_letters"],
+                          'game_score': data["game_score"], 'is_done': data["is_done"],
+                          'is_won': data["is_won"], 'last_saved': datetime.now()}
+                logging.info(fields)
+                function.update_game(username, secret_word, fields)
+                return jsonify(response='OK')
+            except (sqlalchemy.exc.InvalidRequestError, KeyError):
+                return make_response(jsonify(error='Bad request'), 400)
+        else:
+            try:
+                print(data)
+                fields = {'got_hint': data["got_hint"], 'missed_letters': data['missed_letters'],
+                          'correct_letters': data['correct_letters'], 'last_saved': datetime.now()}
+                logging.info(fields)
+                function.update_game(username, secret_word, fields)
+                return jsonify(response='OK')
+            except (sqlalchemy.exc.InvalidRequestError, KeyError):
+                return make_response(jsonify(error='Bad request'), 405)
+
+
+@app.route('/User/<username>/rankings', methods=['GET'])
+def get_all_time_rankings(username):
+    logging.info(f'User with username {username} is checking rankings.')
+    return jsonify(function.get_rankings())
+
+
+@app.route('/User/<username>/last_ten', methods=['GET'])
+def get_last_ten_game_scores(username):
+    return jsonify(function.last_ten_scores(username))
+
+
+@app.route('/User/<username>/top_ten', methods=['GET'])
```

Comment:
The resource that you are fetching in this function and the previous one are "Games" and not user. Therefore, your route should be more like: `@app.route('/Game)/top_ten` and the user can be a query parameter (fetched through request.args) or can be picked from the auth.